### PR TITLE
TEAM TEACHING - overlays and entities, fix to shrunken avatars

### PR DESCRIPTION
**Pull request by ZappoMan**
* make model overlays acutally use sub mesh payload items
* make sure all MOST entities are rendering in pipeline
* fix to shrunken avatars

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 49fa407cd55cdf01d4652d88414a42751d2ba467 available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: bb623eaaedcad0e512efd96b6ea086184d27d6c9 available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6eb9091a01f291322d4243506add73ad8548e483 available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 44382630146dd140045646f372b930b6647bb45a available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: b5b37ce2f534c851699633a5761f20f24ce1de27 available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 8ec2b55ad8873ab8a459b286dbaef0ba3021823f available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f1f48afd61c8cd911c57340d80db99bf624c510a available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by ZappoMan** [comment]
Success!

**Comment by hifi-gustavo** [comment]
Build console output for commit: 54c4cb829aa817499ef2fc6bf794a4a6b8a10d40 available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 54c4cb829aa817499ef2fc6bf794a4a6b8a10d40 available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

File: libraries/entities-renderer/src/EntityTreeRenderer.cpp [hunk 1]
```diff
@@ -493,45 +493,27 @@ void EntityTreeRenderer::render(RenderArgs* renderArgs) {
     if (_tree && !_shuttingDown) {
         renderArgs->_renderer = this;
 
-        checkPendingAddToScene(renderArgs);
-
```
Comment (Atlante45) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: You should remove Model::startScene below too I think

Comment (Atlante45) [inline libraries/entities-renderer/src/EntityTreeRenderer.cpp]: frustum is unused too.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4b389793ad33c3a8b196917993f3a941b68f82f3 available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: c4756f4816e4284c8040291e3276f69ce109d4ec available at: https://gist.github.com/a47a8316fd96b34f21b8<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-5041.dmg

**Pull request merged**